docs/src/accessibility-testing-js.md

```diff
@@ -0,0 +1,292 @@
+expect(violationFingerprints(accessibilityScanResults)).toMatchSnapshot();
+
+// my-test-utils.js
+function violationFingerprints(accessibilityScanResults) {
+    const violationFingerprints = accessibilityScanResults.violations.map(violation => ({
+        rule: violation.id,
+        // These are CSS selectors which uniquely identify each element with
+        // a violation of the rule in question.
+        targets: violation.nodes.map(node => node.target),
+    }));
+
+    return JSON.stringify(violationFingerprints, null, 2);
+}
+```
+
+## Exporting a report as a test attachment
+
+There are many tools and libraries which support generating standalone reports for axe scan results. You can use [`testInfo.attach()`](./api/class-testinfo#test-info-attach) to attach these reports to your test results, which [reporters](./test-reporters) can embed or link as part of your test output.
+
+The following example uses the [`axe-sarif-converter` package](http://npmjs.com/package/axe-sarif-converter) to produce a [SARIF report](https://sarifweb.azurewebsites.net/) and attach it to the test result:
+
+```js
+import { test, expect } from '@playwright/test';
+import { convertAxeToSarif } from 'axe-sarif-converter';
+
+test('example with SARIF report attachment', async function({ page }, testInfo) {
+  await page.goto('https://your-site.com/');
+
+  const accessibilityScanResults = await new AxeBuilder({ page }).analyze();
+
+  const sarifReport = convertAxeToSarif(accessibilityScanResults);
+  await testInfo.attach('sarif-report', {
+    body: JSON.stringify(sarifReport),
+    contentType: 'application/json'
+  });
+
+  expect(accessibilityScanResults.violations).toEqual([]);
+});
+```
+
+## Using a test fixture for common axe configuration
+
+[Test fixtures](./test-fixtures) are a good way to share common `AxeBuilder` configuration across many tests. Some scenarios where this might be useful include:
+* Using a common set of rules among all of your tests
+* Suppressing a known violation in a common element which appears in many different pages
+* Attaching standalone accessibility reports consistently for many scans
+
+The following example demonstrates creating and using a test fixture that covers each of those scenarios.
+
+### Creating a fixture
+
+This example fixture creates an `AxeBuilder` object which is pre-configured with shared `withTags()` and `exclude()` configuration.
+
+```js tab=js-ts
+// axe-test.ts
+import { test as base } from '@playwright/test';
+import AxeBuilder from '@axe-core/playwright';
+
+type AxeFixture = {
+  makeAxeBuilder: () => AxeBuilder;
+};
+
+// Extend base test by providing "makeAxeBuilder"
+//
+// This new "test" can be used in multiple test files, and each of them will get
+// a consistently configured AxeBuilder instance.
+export const test = base.extend<AxeFixture>({
+  makeAxeBuilder: async ({ page }, use, testInfo) => {
+    const makeAxeBuilder = () => new AxeBuilder({ page })
```

Comment:
I think this could be the AxeBuilder instance to make it even easier. WDYT?

```js
const axeBuilder = new AxeBuilder({ page }).....;
await use(axeBuilder);
```

Or, if you make it a function, let it have some options that I can pass from a test. Do you think that makes sense?

Reply:
The original version of this example did actually provide an `AxeBuilder` instance directly instead of a factory function, but when I tested out the code in a working project, I found that doing that led to the following gotcha during usage:

```ts
test('example fixture usage with multiple scans per test', async ({ myPageModel, axeBuilder }) => {
  const dialogResults = await axeBuilder
    .include('#welcome-dialog')
    .disableRules(['color-contrast']) // known issue #123
    .analyze();
  expect(dialogResults.violations).toEqual([]);

  await myPageModel.dismissWelcomeDialog();

  // Oops: this looks like it's doing a new scan, but actually `color-contrast` is still disabled because it's the same axeBuilder instance
  const mainContentResults = await axeBuilder
    .include('main')
    .analyze();
  expect(mainContentResults.violations).toEqual([]);
});
```

I wanted the example fixture to be a pit of success, so I thought it felt a bit too dangerous to have the easiest way to run multiple scans per test lead to that gotcha. I considered explaining the reasoning behind using a factory function instead of a direct AxeBuilder within the example, but it felt like it required so much space to explain reasonably that it took away from the rest of the example.

As far as providing options, in general the factory still returns an axeBuilder instance and the usage example demonstrates how to apply additional options on top of it from within a test. Are you asking about providing options that pass-through to builder methods, or something else?

Reply:
Great point on the reuse, I didn't realize that! Factory it is then.

> Are you asking about providing options that pass-through to builder methods, or something else?

I thought maybe you have any ideas for the options, because I don't have anything specific in mind. But I guess if any option comes up later, we'll easily add it. Sorry for the noise.

Reply:
No worries, I appreciate the feedback! I definitely agree that if it weren't for the gotcha, avoiding the unnecessary layer of factory would be cleaner/better.